std-bits/base/src/main/java/org/enso/base/ObjectComparator.java

```diff
@@ -1,73 +1,69 @@
 package org.enso.base;
 
+import org.graalvm.polyglot.Context;
 import org.graalvm.polyglot.Value;
 
 import java.time.LocalDate;
 import java.time.LocalTime;
 import java.time.ZonedDateTime;
 import java.util.Comparator;
 import java.util.Locale;
-import java.util.function.BiFunction;
 import java.util.function.Function;
+import java.util.function.BiFunction;
 
 public class ObjectComparator implements Comparator<Object> {
-  private static ObjectComparator INSTANCE;
-
-  /**
-   * A singleton instance of an ObjectComparator.
-   *
-   * @param fallbackComparator this MUST be the default .compare_to function for Enso. Needs to be
-   *     passed to allow calling back from Java.
-   * @return Comparator object.
-   */
-  public static ObjectComparator getInstance(Function<Object, Function<Object, Value>> fallbackComparator) {
-    if (INSTANCE == null) {
-      INSTANCE = new ObjectComparator(fallbackComparator);
+  public static final ObjectComparator DEFAULT = new ObjectComparator();
+
+  private static BiFunction<Object, Object, Integer> EnsoCompareCallback = null;
+
+  private static void initCallbacks() {
+    if (EnsoCompareCallback == null) {
+      var module = Context.getCurrent().getBindings("enso").invokeMember("get_module", "Standard.Base.Data.Ordering");
+      var type = module.invokeMember("get_type", "Comparable");
+
+      var are_equal = module.invokeMember("get_method", type, "compare_callback");
+      EnsoCompareCallback = (v, u) -> {
+        var result = are_equal.execute(null, v, u);
+        if (result.isNull()) {
+          throw new CompareException(u, v);
+        } else {
+          return result.asInt();
+        }
+      };
     }
-
-    return INSTANCE;
   }
 
-  private final Function<Object, Function<Object, Value>> fallbackComparator;
-  private final Function<String, Function<String, Value>> textComparator;
-
-
-  public ObjectComparator() {
-    this(
-        (a) -> (b) -> {
-          throw new CompareException(a, b);
-        });
+  public static int ensoCompare(Object value, Object other) throws ClassCastException {
```

Comment:
Shouldn't there be rather `CompareException` in the signature? Note that recently, I have modified this behavior a bit - instead of throwing `ClassCastException` for invalid comparison, it throws `CompareException`.